src/widgets/Jitsi.ts

```diff
@@ -36,33 +34,21 @@ export class Jitsi {
         return this.domain || 'jitsi.riot.im';
     }
 
-    constructor() {
-        // We rely on the first call to be an .update() instead of doing one here. Doing one
-        // here could result in duplicate calls to the homeserver.
-
-        // Start a timer to update the server info regularly
-        setInterval(() => this.update(), JITSI_WK_CHECK_INTERVAL);
-    }
-
     public async update(): Promise<any> {
         // Start with a default of the config's domain
         let domain = (SdkConfig.get()['jitsi'] || {})['preferredDomain'] || 'jitsi.riot.im';
 
-        // Now request the .well-known config to see if it changed
-        if (MatrixClientPeg.get()) {
-            try {
-                console.log("Attempting to get Jitsi conference information from homeserver");
-
-                const homeserverDomain = MatrixClientPeg.getHomeserverName();
-                const discoveryResponse = await AutoDiscovery.getRawClientConfig(homeserverDomain);
-                if (discoveryResponse && discoveryResponse[JITSI_WK_PROPERTY]) {
-                    const wkPreferredDomain = discoveryResponse[JITSI_WK_PROPERTY]['preferredDomain'];
-                    if (wkPreferredDomain) domain = wkPreferredDomain;
-                }
-            } catch (e) {
-                // These are non-fatal errors
-                console.error(e);
-            }
+        const cli = MatrixClientPeg.get();
+        const discoveryResponse = cli && MatrixClientPeg.get().getClientWellKnown();
+
+        if (cli) {
+            cli.on("WellKnown.client", () => this.update);
+        }
```

Comment:
The update method is adding itself as a listener? How does it start listening in the first place, and does it leak listeners after that?

Reply:
True about the leakiness, the first update call is from Lifecycle, will split into start & update